packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -276,7 +278,13 @@ export class TokenRatesController extends PollingControllerV1<
     const detectedTokens =
       allDetectedTokens[chainId]?.[this.config.selectedAddress] || [];
 
-    return [...tokens, ...detectedTokens].map((token) => toHex(token.address));
+    return [
+      ...new Set(
+        [...tokens, ...detectedTokens].map((token) =>
+          toHex(toChecksumHexAddress(token.address)),
```

Comment:
is the `toHex` call redundant alongside `toChecksumHexAddress`?

Reply:
It's not — unfortunately `toChecksumHexAddress` (located in `controller-utils`) doesn't use a type predicate to assert that its return value is a `Hex`. `toHex` (which is from `@metamask/utils`) does, though. So it's redundant in the sense that it doesn't do anything, but it does convert this to a `Hex`.